src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1232,19 +1238,66 @@ pub struct QmpSocket {
 }
 
 impl QmpSocket {
-    fn new() -> Result<Vec<Self>> {
-        let qmp_sockets: Vec<QmpSocket> = Vec::new();
-        let _proto = MonitorProtocol::new("qmp");
+    fn new(sid: &str, extra_monitor: bool) -> Result<Vec<Self>> {
```

Comment:
I guess this works, however we now have a constructor that returns a container of instances instead of an instance - which is kind of unexpected.  :-)  Honestly I'd prefer if the `QmpSocket::new()` took the monitor protocol as a parameter and then returned a single instance as usual.

Then `QmpSocket::fd` and `QmpSocket::name` had probably better become a single member of an enum type since only one of them is ever needed.

Also, `QemuCmdLine::new()` would unconditionally create a `QmpSocket` instance for the Qmp socket and optionally, based on hypervisor config, another instance for Hmp, similar to how MMIO and RTC are already handled.